src/librawspeed/decoders/IiqDecoder.cpp

```diff
@@ -187,6 +199,9 @@ RawImage IiqDecoder::decodeRawInternal() {
   mRaw->createData();
 
```

Comment:
```diff
// FIXME: could be wrong. max "active pixels" in "Sensor+" mode - "101 MP"
if (width == 0 || height == 0 || width > 11608 || height > 8708)
  ThrowRDE("Unexpected image dimensions found: (%u; %u)", width, height);

+if (split_col > width || split_row > height)
+  ThrowRDE("Invalid sensor quadrants: (%u; %u)", split_col, split_row);
```
or something like that

Reply:
Also, what happens if
* `correction_meta_data` is there, but not `split_col` and not `split_row`?
  Does that mean it is all one quadrant?
* `correction_meta_data` is not there, but `split_col` and/or `split_row` is there?

I.e. needs more sanitization.

Reply:
I would imagine it's possible for the one to be present without the other, because there's a range of sensor corrections and only some of them rely on a split column/row.  Some of them could even apply to images with more than just four segments, in which case split_col and split_row would be nonsensical, and the camera might include them just for informational purposes even if there's no corrections needed (unfortunately I don't have any cameras without corrections to test this), so I don't think there's anything I can really infer from the presence/absence of one or the other